<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PaletteEditor</class>
 <widget class="QWidget" name="PaletteEditor">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>294</width>
    <height>330</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Palettes</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_2">
   <item row="0" column="0">
    <widget class="QFrame" name="frame">
     <property name="minimumSize">
      <size>
       <width>278</width>
       <height>16</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>330</width>
       <height>16777215</height>
      </size>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Sunken</enum>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QFrame" name="frame_2">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <layout class="QGridLayout">
         <property name="margin">
          <number>0</number>
         </property>
         <property name="spacing">
          <number>0</number>
         </property>
         <item row="0" column="0">
          <widget class="QLabel" name="m_paletteLabel">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
             <horstretch>0</horstretch>
             <verstretch>1</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>256</width>
             <height>60</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>256</width>
             <height>60</height>
            </size>
           </property>
           <property name="frameShape">
            <enum>QFrame::StyledPanel</enum>
           </property>
           <property name="frameShadow">
            <enum>QFrame::Sunken</enum>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QTabWidget" name="m_tabWidget">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>1</verstretch>
         </sizepolicy>
        </property>
        <property name="tabShape">
         <enum>QTabWidget::Rounded</enum>
        </property>
        <property name="currentIndex">
         <number>0</number>
        </property>
        <widget class="QWidget" name="m_builtinTab">
         <attribute name="title">
          <string>Palettes</string>
         </attribute>
         <layout class="QHBoxLayout">
          <item>
           <widget class="QListView" name="m_palettesView">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="iconSize">
             <size>
              <width>256</width>
              <height>30</height>
             </size>
            </property>
            <property name="viewMode">
             <enum>QListView::IconMode</enum>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="m_browseTab">
         <attribute name="title">
          <string>Browse</string>
         </attribute>
         <layout class="QVBoxLayout" name="_3">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_5">
            <item>
             <widget class="QLineEdit" name="m_browseLineEdit"/>
            </item>
            <item>
             <widget class="QToolButton" name="m_openButton">
              <property name="text">
               <string>...</string>
              </property>
              <property name="icon">
               <iconset resource="../qosmic.qrc">
                <normaloff>:/icons/silk/folder.xpm</normaloff>:/icons/silk/folder.xpm</iconset>
              </property>
              <property name="iconSize">
               <size>
                <width>18</width>
                <height>18</height>
               </size>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QListView" name="m_browseView">
            <property name="iconSize">
             <size>
              <width>256</width>
              <height>30</height>
             </size>
            </property>
            <property name="viewMode">
             <enum>QListView::IconMode</enum>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="m_gradientsTab">
         <attribute name="title">
          <string>Edit</string>
         </attribute>
         <layout class="QGridLayout" name="_2" rowstretch="0,0,0,0,1">
          <item row="0" column="0">
           <layout class="QHBoxLayout" name="horizontalLayout">
            <item>
             <widget class="QLabel" name="m_rotateLabel">
              <property name="text">
               <string>rotate</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSlider" name="m_rotateSlider">
              <property name="maximum">
               <number>255</number>
              </property>
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="1" column="0">
           <layout class="QGridLayout" name="gradientsTabGridLayout" rowstretch="0,0,1,0" rowminimumheight="0,0,60,0">
            <item row="0" column="0">
             <widget class="QGroupBox" name="m_gradientSpreadGroupBox">
              <property name="title">
               <string/>
              </property>
              <layout class="QHBoxLayout" name="horizontalLayout_2">
               <property name="spacing">
                <number>0</number>
               </property>
               <property name="margin">
                <number>0</number>
               </property>
               <item>
                <widget class="QRadioButton" name="m_padSpreadButton">
                 <property name="text">
                  <string>pad</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QRadioButton" name="m_repeatSpreadButton">
                 <property name="text">
                  <string>repeat</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QRadioButton" name="m_reflectSpreadButton">
                 <property name="text">
                  <string>reflect</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="GradientStopsEditor" name="m_gradientEnds" native="true">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
                <horstretch>0</horstretch>
                <verstretch>1</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>256</width>
                <height>10</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>256</width>
                <height>10</height>
               </size>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="m_gradientLabel">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>2</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>256</width>
                <height>0</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>256</width>
                <height>60</height>
               </size>
              </property>
              <property name="frameShape">
               <enum>QFrame::StyledPanel</enum>
              </property>
              <property name="frameShadow">
               <enum>QFrame::Sunken</enum>
              </property>
             </widget>
            </item>
            <item row="3" column="0">
             <widget class="GradientStopsEditor" name="m_gradientStops" native="true">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
                <horstretch>0</horstretch>
                <verstretch>1</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>256</width>
                <height>10</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>256</width>
                <height>10</height>
               </size>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="3" column="0">
           <layout class="QHBoxLayout" name="horizontalLayout_3">
            <item>
             <widget class="IntValueEditor" name="m_randomGradientSpinBox">
              <property name="toolTip">
               <string>number of randomly generated stops,
&lt; 2 implies random number on [2,128]</string>
              </property>
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>128</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QToolButton" name="m_randomGradientButton">
              <property name="toolTip">
               <string>random gradient</string>
              </property>
              <property name="text">
               <string>new</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QToolButton" name="m_resetGradientButton">
              <property name="toolTip">
               <string>reset</string>
              </property>
              <property name="text">
               <string>reset</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QToolButton" name="m_saveGradientButton">
              <property name="toolTip">
               <string>save</string>
              </property>
              <property name="text">
               <string>save</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="4" column="0">
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="2" column="0">
           <spacer name="horizontalSpacer_2">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>8</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>IntValueEditor</class>
   <extends>QSpinBox</extends>
   <header>intvalueeditor.h</header>
  </customwidget>
  <customwidget>
   <class>GradientStopsEditor</class>
   <extends>QWidget</extends>
   <header>gradientstopseditor.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../qosmic.qrc"/>
 </resources>
 <connections/>
</ui>
